### PR TITLE
Remove deactivated currencies from new CLDR

### LINKED ISSUE #13270: Currency list isn't the same as in 1.7.5.1
**Describe the bug**
Currency list on 1.7.6.x isn't the same as 1.7.5.1

**To Reproduce**
1. Go to International > Localization > Currencies
2. Click on Add new currency
3. Look the list

For example, in 1.7.5.1 I had ZMW
In 1.7.6.0, I don't have it anymore but I have ZWD, ZWL, ZWR

capture d'écran_1345

capture d'écran_1344

**Additionnal information**
PrestaShop version: 1.7.6.X
PHP version: N/A

### PR DESCRIPTION
**Pull request by tomlev**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | some currencies do not need to be added if it is deactivated for a long time. Deprecate adding currencies BYR, LTL, STD which are deprecated
| Type?         | bug fix
| Category?     |  BO
| BC breaks?    | yes
| Deprecations? | yes
| Fixed ticket? | fixes #13270
| How to test?  | in BO go to currency add, the dropdown should show only active currency. CLDR uses file cache, so delete following directory if you try to test between several versions: /var/cache/dev

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13367)

### CODE REVIEW

File: src/Core/Localization/CLDR/Reader.php [hunk 1]
```diff
@@ -509,11 +516,36 @@ protected function mapLocaleData(SimpleXMLElement $xmlLocaleData, $supplementalD
...
+                // check if currency is still active in one territory
+                $isActiveSomewhere = false;
+                $currencyDates = $this->supplementalXml->supplementalData->xpath(
+                    '//region/currency[@iso4217="' . $currencyCode . '"]'
+                );
+                if (!empty($currencyDates)) {
+                    foreach ($currencyDates as $currencyDate) {
+                        if (empty($currencyDate->attributes()->to)) {
+                            // no date "to": currency is active in some territory
+                            $isActiveSomewhere = true;
```
Comment (Quetzacoalt91) [inline src/Core/Localization/CLDR/Reader.php]: I don't know how many $currencyDate you may have, but having this part of the code in a separate method could allow you to return as soon as you can set the value as `true`

File: src/Core/Localization/CLDR/Reader.php [hunk 2]
```diff
@@ -509,11 +516,36 @@ protected function mapLocaleData(SimpleXMLElement $xmlLocaleData, $supplementalD
...
             $currencies = $localeData->getCurrencies();
             foreach ($currencyData->currency as $currencyNode) {
                 $currencyCode = (string) $currencyNode['type'];
 
+                // check if currency is still active in one territory
+                $isActiveSomewhere = false;
+                $currencyDates = $this->supplementalXml->supplementalData->xpath(
+                    '//region/currency[@iso4217="' . $currencyCode . '"]'
+                );
+                if (!empty($currencyDates)) {
```
Comment (Quetzacoalt91) [inline src/Core/Localization/CLDR/Reader.php]: You could directly `continue;` if the $currencyDates is empty. There is no chance you set it as true further. :)

File: src/Core/Localization/CLDR/Reader.php (hunk 1 above)
Comment (tomlev) [reply src/Core/Localization/CLDR/Reader.php]: indeed ! (one $currencyDate for each territory the currency is or was used)

File: src/Core/Localization/CLDR/Reader.php (hunk 2 above)
Comment (tomlev) [reply src/Core/Localization/CLDR/Reader.php]: done

File: src/Core/Localization/CLDR/Reader.php [hunk 3]
```diff
@@ -597,4 +608,57 @@ protected function getDigitsData()
...
+        if ($currencyCode == self::CURRENCY_CODE_TEST) {
+            return false;
+        }
+        // check if currency is still active in one territory
+        $currencyDates = $supplementalData->xpath('//region/currency[@iso4217="' . $currencyCode . '"]');
+        if (empty($currencyDates)) {
+            // no territory with dates means currency was never used
+            return false;
+        }
+        if (!$this->isCurrencyActiveSomewhere($currencyDates, $currencyActiveDateThreshold)) {
```
Comment (Quetzacoalt91) [inline src/Core/Localization/CLDR/Reader.php]:
```suggestion
        return $this->isCurrencyActiveSomewhere($currencyDates, $currencyActiveDateThreshold);
```

Comment (Quetzacoalt91) [reply src/Core/Localization/CLDR/Reader.php]: The next lines can be removed.

File: src/Core/Localization/CLDR/CurrencyData.php [hunk 4]
```diff
@@ -222,4 +233,22 @@ public function setSymbols($symbols)
 
         return $this;
     }
+
+    /**
+     * is currency still active in some territory
+     *
+     * @return bool
```
Comment (Quetzacoalt91) [inline src/Core/Localization/CLDR/CurrencyData.php]: Looks like it can return `null` too.

**Review by Quetzacoalt91 — APPROVED** [review]

**Pull request merged**